### PR TITLE
CLOUDTRUST-3443: add key ID in database where encryption is used

### PR DESCRIPTION
**Pull request by lagess**

### CODE REVIEW

File: Gopkg.toml [hunk 1]
```diff
@@ -25,9 +25,13 @@
 #   unused-packages = true
 
 
-[[constraint]]
+[[override]]
   name = "github.com/cloudtrust/common-service"
-  version = "2.4.3"
+  branch = "CLOUDTRUST-3443-key-rotation"
```
Comment (fperot74) [inline Gopkg.toml:30]: Don't forget to update before merging

File: configs/keycloak_bridge.yml [hunk 2]
```diff
@@ -7,7 +7,6 @@ management-http-host-port: 0.0.0.0:8877
 account-http-host-port: 0.0.0.0:8866
 register-http-host-port: 0.0.0.0:8855
 mobile-http-host-port: 0.0.0.0:8844
-monitoring-http-host-port: 0.0.0.0:8899
```
Comment (fperot74) [inline configs/keycloak_bridge.yml:10]: Why is this line removed?

File: configs/keycloak_bridge.yml [hunk 3]
```diff
@@ -49,10 +56,10 @@ mobile-audience-required: "mobile"
 validation-basic-auth-token: "idnowsuperpasswordverylongandstrong"
 
 # DB encryption key
-db-aesgcm-key: oYP5DhsaW8dLtBt89i9cvXqz+zQTJBHWdFejLWLN/28=
+db-aesgcm-key: '[{"kid":"KBB_4","value":"AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAB="},{"kid":"KBB_3","value":"AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA="}]'
```
Comment (fperot74) [inline configs/keycloak_bridge.yml:59]: AAAAAA....AB: I would set BBBBBBBBB....BB so that the difference is more visual

File: internal/keycloakb/archivedbmodule_test.go [hunk 4]
```diff
@@ -31,21 +31,21 @@ func TestStoreUserDetails(t *testing.T) {
 	var ctx = context.TODO()
 
 	t.Run("Encryption fails", func(t *testing.T) {
-		mockCrypter.EXPECT().Encrypt(gomock.Any(), userIDBytes).Return(nil, anyError)
+		mockCrypter.EXPECT().Encrypt(gomock.Any(), userIDBytes).Return(nil, "", anyError)
```
Comment (fperot74) [inline internal/keycloakb/archivedbmodule_test.go:34]: It would be better to avoid empty key names even if it is not used after that

File: internal/keycloakb/usersdbmodule_test.go [hunk 5]
```diff
@@ -23,24 +23,24 @@ func TestStoreOrUpdateUserDetails(t *testing.T) {
 	var userID = "123789"
 	t.Run("Update succesful", func(t *testing.T) {
 
-		mockDB.EXPECT().Exec(gomock.Any(), "realmId", &userID, gomock.Any(), gomock.Any()).Return(nil, nil).Times(1)
+		mockDB.EXPECT().Exec(gomock.Any(), "realmId", &userID, gomock.Any(), gomock.Any(), gomock.Any()).Return(nil, nil).Times(1)
 		var configDBModule = NewUsersDetailsDBModule(mockDB, mockCrypter, log.NewNopLogger())
-		mockCrypter.EXPECT().Encrypt(gomock.Any(), gomock.Any()).Return(nil, nil).Times(1)
+		mockCrypter.EXPECT().Encrypt(gomock.Any(), gomock.Any()).Return(nil, "", nil).Times(1)
```
Comment (fperot74) [inline internal/keycloakb/usersdbmodule_test.go:28]: Empty key name?

**Review by fperot74 — APPROVED** [review]

File: configs/keycloak_bridge.yml (hunk 2 above)
Comment (lagess) [reply configs/keycloak_bridge.yml:10]: The content of the files comes from the dev-env --> the monitoring is not listed in this config file.
I'll check with @harture and fix where appropriate.

File: configs/keycloak_bridge.yml [hunk 6]
```diff
@@ -62,19 +70,17 @@ trustid-groups:
...
+  default: http://keycloak.local:8080
 keycloak-timeout: 5s
 
 # DB Audit RW
 # db-audit-rw-enabled was previously named events-db
 db-audit-rw-enabled: false
-db-audit-rw-host-port: 127.0.0.1:3306
-db-audit-rw-username: bridge
-db-audit-rw-password: bridge-password
+db-audit-rw-host-port: localhost:3306
```
Comment (harture) [inline configs/keycloak_bridge.yml:81]: We must use host.docker.internal to be compatible with docker container in dev env

File: internal/keycloakb/archivedbmodule.go [hunk 7]
```diff
@@ -45,13 +45,13 @@ func (a *archiveDBModule) StoreUserDetails(ctx context.Context, realm string, us
 	}
 
 	// encrypt the JSON containing the details on the user
-	encryptedData, err := a.cipher.Encrypt(details, []byte(*user.ID))
+	encryptedData, keyId, err := a.cipher.Encrypt(details, []byte(*user.ID))
```
Comment (harture) [inline internal/keycloakb/archivedbmodule.go:48]: I think it would be better to have a separate method to retrieve the currently used keyId. (See comment in common-service)

**Review by harture — APPROVED** [review]
